server/services/WalletService.js

```diff
@@ -0,0 +1,24 @@
+const WalletRepository = require("../repositories/WalletRepository");
+const Wallet = require("../models/Wallet");
+const expect = require("expect-runtime");
+
+class WalletService{
+  constructor(){
+    this.walletRepository = new WalletRepository();
+  }
+
+  async getById(id){
+    const object = await this.walletRepository.getById(id);
+    const wallet = new Wallet(object.id);
+    return wallet;
+  }
+
+  async getByName(name){
+    const object = await this.walletRepository.getByName(name);
+    expect(object).match({id:expect.any(Number)});
```

Comment:
We have this check here for the id number type, but we also have the check in the wallet model constructor. I think it is sufficient to just check it here before we invoke the constructor (so also add this check before line 12), or only check it in the constructor and not here. What do you think?

Reply:
@mzyau sorry about some confusion about the usage of `expect`, don't these codes (runtime expect checking) too seriously, you can remove them at any place you find them if you don't like it. Zaven thinks it's overkill to check types at runtime, but I just use it to make coding more comfortable and easier. 
let the wrong type be detected earlier make it's easier to check what's the problem with the code. 
Especially when we mock module in unit test, it's easy to make mistake on the response type, watch code here, if mock the walletRepository, sometimes the sinon.stub return wrong type for this, so I added a check here, this wrongdoing happens, especially when we need to refactor or change some input/output on a function.

Reply:
Got it! If it's just for debugging then it's not a big deal